rasa/engine/graph.py

```diff
@@ -436,10 +440,27 @@ def __call__(
         Returns:
             The node name and its output.
         """
-        received_inputs: Dict[Text, Any] = dict(inputs_from_previous_nodes)
+        # filter out arguments that dask couldn't lookup
+        received_inputs: Dict[Text, Any] = {}
+        for i in inputs_from_previous_nodes:
+            if isinstance(i, tuple):
+                arg_name, arg_value = i
+                received_inputs[arg_name] = arg_value
+            else:
+                logger.warning(
+                    f"Received unresolved argument: '{i}'. "
+                    f"Another component should have provided this as an "
+                    f"output."
+                )
 
         kwargs = {}
         for input_name, input_node in self._inputs.items():
+            if input_node not in received_inputs:
+                raise GraphRunError(
+                    f"Missing input to run node '{self._node_name}'. "
+                    f"Expected input '{input_node}' for parameter "
+                    f"'{input_name}'."
+                )
```

Comment:
Maybe also some potential for improving the naming and the error message here.
 I think `input_node` should probably be `input_provider_node_name` and then in the second part of the error message you could say "Expected '`input_provider_node_name`' to provide parameter '`input_name`'."